schema-lending.graphql

```diff
@@ -141,18 +171,151 @@ type InterestRate @entity {
   type: InterestRateType!
 }
 
+# There is a wide range of formats of lending protocols
+# And there are many unique mappings when it comes to input / output tokens
+# This generic type will help define the unique relationships between tokens in a market
+# You can think of this as a 1:M relationship between 1 input token and output tokens
+type TokenData @entity {
+  " { Market Address }-{ Input Token Address }-{ Optional: Snapshot ID } "
+  id: Bytes!
+
+  " Can you use the output token as collateral "
+  canUseAsCollateral: Boolean!
+
+  " Maximum loan-to-value ratio as a percentage value (e.g. 75% for DAI in Aave) "
+  maximumLTV: BigDecimal!
+
+  " Liquidation threshold as a percentage value (e.g. 80% for DAI in Aave). When it is reached, the position is defined as undercollateralised and could be liquidated "
+  liquidationThreshold: BigDecimal!
+
+  " Liquidation penalty (or the liquidation bonus for liquidators) as a percentage value. It is the penalty/bonus price on the collateral when liquidators purchase it as part of the liquidation of a loan that has passed the liquidation threshold "
+  liquidationPenalty: BigDecimal!
+
+  " Can the user choose to isloate assets in this market. e.g. only this market's collateral can be used for a borrow in Aave V3 "
+  canIsolate: Boolean!
+
+  ##### Token Data #####
+
+  " Tokens that need to be deposited in this market to take a position in protocol (should be alphabetized) "
+  inputToken: Token!
+
+  " Amount of input tokens in the market (same order as inputTokens) "
+  inputTokenBalance: BigInt!
+
+  " Prices in USD of the input tokens (same order as inputTokens) "
+  inputTokenPricesUSD: BigDecimal!
+
+  " Tokens that are minted to track ownership of position in protocol (e.g. aToken, cToken). Leave as null if doesn't exist (should be alphabetized) "
+  outputTokens: [Token!]
+
+  " Total supply of output tokens (same order as outputTokens) "
+  outputTokenBalances: [BigInt!]
```

Comment:
-> outputTokenSupply?

Reply:
I wanted to keep the same theme with inputTokenBalance. (and `outputTokenSupplies` looks kinda strange to me)

Do you have more thoughts?

Reply:
I kinda prefer supply 😅 

Since balances refer to the actual balance held at the market, and supply means the amount of outputTokens minted, which are not held in the market but in arbitrary wallets